From 05aa9716fe7e9baac49ce50e3aca8d8000894cfc Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Tue, 4 Jul 2017 14:52:12 +0800
Subject: [PATCH 2048/5242] MLK-15330-2: ARM64: imx8qm/qxp: modify all device
 nodes using edmav3

commit  5e6d2c63f98c26ce13dc8e0589e8a400fd3c5d02 from
https://source.codeaurora.org/external/imx/linux-imx.git

Modify all device nodes which use edmav3,since dma-cell down from 4 to
3.

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi      |   30 ++++++++++----------
 .../boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts |    2 +-
 arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi     |   24 ++++++++--------
 3 files changed, 28 insertions(+), 28 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
index 00aee41..b0b2a5d 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
@@ -1688,8 +1688,8 @@
 		assigned-clock-rates = <80000000>;
 		power-domains = <&pd_dma_lpuart1>;
 		dma-names = "tx","rx";
-		dmas = <&edma0 15 0 0 0>,
-			<&edma0 14 0 1 0>;
+		dmas = <&edma0 15 0 0>,
+			<&edma0 14 0 1>;
 		status = "disabled";
 	};
 
@@ -1705,8 +1705,8 @@
 		assigned-clock-rates = <80000000>;
 		power-domains = <&pd_dma_lpuart2>;
 		dma-names = "tx","rx";
-		dmas = <&edma0 17 0 0 0>,
-			<&edma0 16 0 1 0>;
+		dmas = <&edma0 17 0 0>,
+			<&edma0 16 0 1>;
 		status = "disabled";
 	};
 
@@ -1722,8 +1722,8 @@
 		assigned-clock-rates = <80000000>;
 		power-domains = <&pd_dma_lpuart3>;
 		dma-names = "tx","rx";
-		dmas = <&edma0 19 0 0 0>,
-			<&edma0 18 0 1 0>;
+		dmas = <&edma0 19 0 0>,
+			<&edma0 18 0 1>;
 		status = "disabled";
 	};
 
@@ -1739,8 +1739,8 @@
 		assigned-clock-rates = <80000000>;
 		power-domains = <&pd_dma_lpuart4>;
 		dma-names = "tx","rx";
-		dmas = <&edma0 21 0 0 0>,
-			<&edma0 20 0 1 0>;
+		dmas = <&edma0 21 0 0>,
+			<&edma0 20 0 1>;
 		status = "disabled";
 	};
 
@@ -1756,7 +1756,7 @@
 		      <0x0 0x5a330000 0x0 0x10000>, /* channel19 UART3 tx */
 		      <0x0 0x5a340000 0x0 0x10000>, /* channel20 UART4 rx */
 		      <0x0 0x5a350000 0x0 0x10000>; /* channel21 UART4 tx */
-		#dma-cells = <4>;
+		#dma-cells = <3>;
 		dma-channels = <10>;
 		interrupts = <GIC_SPI 434 IRQ_TYPE_LEVEL_HIGH>,
 			     <GIC_SPI 435 IRQ_TYPE_LEVEL_HIGH>,
@@ -1790,7 +1790,7 @@
 			<0x0 0x59290000 0x0 0x10000>, /* spdif0 tx */
 			<0x0 0x592c0000 0x0 0x10000>, /* sai0 rx */
 			<0x0 0x592d0000 0x0 0x10000>; /* sai0 tx */
-		#dma-cells = <4>;
+		#dma-cells = <3>;
 		shared-interrupt;
 		dma-channels = <12>;
 		interrupts = <GIC_SPI 374 IRQ_TYPE_LEVEL_HIGH>, /* asrc0 */
@@ -2101,7 +2101,7 @@
 			<&clk IMX8QM_CLK_DUMMY>,
 			<&clk IMX8QM_CLK_DUMMY>;
 		clock-names = "core", "extal", "fsys", "spba";
-		dmas = <&edma2 6 0 1 0>, <&edma2 7 0 0 0>;
+		dmas = <&edma2 6 0 1>, <&edma2 7 0 0>;
 		dma-names = "rx", "tx";
 		power-domains = <&pd_esai0>;
 		status = "disabled";
@@ -2127,7 +2127,7 @@
 			      "rxtx3", "rxtx4",
 			      "rxtx5", "rxtx6",
 			      "rxtx7", "spba";
-		dmas = <&edma2 8 0 1 0>, <&edma2 9 0 0 0>;
+		dmas = <&edma2 8 0 1>, <&edma2 9 0 0>;
 		dma-names = "rx", "tx";
 		power-domains = <&pd_spdif0>;
 		status = "disabled";
@@ -2144,7 +2144,7 @@
 			<&clk IMX8QM_CLK_DUMMY>;
 		clock-names = "bus", "mclk0", "mclk1", "mclk2", "mclk3";
 		dma-names = "rx", "tx";
-		dmas = <&edma2 12 0 1 0>, <&edma2 13 0 0 0>;
+		dmas = <&edma2 12 0 1>, <&edma2 13 0 0>;
 		status = "disabled";
 		power-domains = <&pd_sai0>;
 	};
@@ -2179,8 +2179,8 @@
 			"asrck_8", "asrck_9", "asrck_a", "asrck_b",
 			"asrck_c", "asrck_d", "asrck_e", "asrck_f",
 			"spba";
-		dmas = <&edma2 0 0 0 0>, <&edma2 1 0 0 0>, <&edma2 2 0 0 0>,
-			<&edma2 3 0 1 0>, <&edma2 4 0 1 0>, <&edma2 5 0 1 0>;
+		dmas = <&edma2 0 0 0>, <&edma2 1 0 0>, <&edma2 2 0 0>,
+			<&edma2 3 0 1>, <&edma2 4 0 1>, <&edma2 5 0 1>;
 		dma-names = "rxa", "rxb", "rxc",
 				"txa", "txb", "txc";
 		fsl,asrc-rate  = <8000>;
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts
index 9303238..c0d1bd1 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts
@@ -96,7 +96,7 @@
 			<&clk IMX8QXP_AUD_ESAI_0_EXTAL_IPG>;
 	assigned-clock-parents = <&clk IMX8QXP_AUD_ACM_AUD_PLL_CLK0_CLK>;
 	assigned-clock-rates = <0>, <786432000>, <24576000>, <24576000>, <24576000>;
-	dmas = <&edma2 23 0 1 1>, <&edma2 21 0 0 1>;
+	dmas = <&edma2 23 0 3>, <&edma2 21 0 2>;
 	status = "okay";
 };
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
index 13cb35d4..52a9d16 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
@@ -1222,8 +1222,8 @@
 		assigned-clock-rates = <80000000>;
 		power-domains = <&pd_dma_lpuart1>;
 		dma-names = "tx","rx";
-		dmas = <&edma0 11 0 0 0>,
-			<&edma0 10 0 1 0>;
+		dmas = <&edma0 11 0 0>,
+			<&edma0 10 0 1>;
 		status = "disabled";
 	};
 
@@ -1239,8 +1239,8 @@
 		assigned-clock-rates = <80000000>;
 		power-domains = <&pd_dma_lpuart2>;
 		dma-names = "tx","rx";
-		dmas = <&edma0 13 0 0 0>,
-			<&edma0 12 0 1 0>;
+		dmas = <&edma0 13 0 0>,
+			<&edma0 12 0 1>;
 		status = "disabled";
 	};
 
@@ -1256,8 +1256,8 @@
 		assigned-clock-rates = <80000000>;
 		power-domains = <&pd_dma_lpuart3>;
 		dma-names = "tx","rx";
-		dmas = <&edma0 15 0 0 0>,
-			<&edma0 14 0 1 0>;
+		dmas = <&edma0 15 0 0>,
+			<&edma0 14 0 1>;
 		status = "disabled";
 	};
 
@@ -1271,7 +1271,7 @@
 		      <0x0 0x5a2d0000 0x0 0x10000>, /* channel13 UART2 tx */
 		      <0x0 0x5a2e0000 0x0 0x10000>, /* channel14 UART3 rx */
 		      <0x0 0x5a2f0000 0x0 0x10000>; /* channel15 UART3 tx */
-		#dma-cells = <4>;
+		#dma-cells = <3>;
 		dma-channels = <8>;
 		interrupts = <GIC_SPI 434 IRQ_TYPE_LEVEL_HIGH>,
 			     <GIC_SPI 435 IRQ_TYPE_LEVEL_HIGH>,
@@ -1304,7 +1304,7 @@
 			<0x0 0x592d0000 0x0 0x10000>, /* sai0 tx */
 			<0x0 0x59350000 0x0 0x10000>,
 			<0x0 0x59370000 0x0 0x10000>;
-		#dma-cells = <4>;
+		#dma-cells = <3>;
 		shared-interrupt;
 		dma-channels = <14>;
 		interrupts = <GIC_SPI 374 IRQ_TYPE_LEVEL_HIGH>, /* asrc 0 */
@@ -1350,7 +1350,7 @@
 			<&clk IMX8QXP_CLK_DUMMY>;
 		clock-names = "bus", "mclk0", "mclk1", "mclk2", "mclk3";
 		dma-names = "rx", "tx";
-		dmas = <&edma2 12 0 1 0>, <&edma2 13 0 0 0>;
+		dmas = <&edma2 12 0 1>, <&edma2 13 0 0>;
 		status = "disabled";
 		power-domains = <&pd_sai0>;
 	};
@@ -1385,8 +1385,8 @@
 			"asrck_8", "asrck_9", "asrck_a", "asrck_b",
 			"asrck_c", "asrck_d", "asrck_e", "asrck_f",
 			"spba";
-		dmas = <&edma2 0 0 0 0>, <&edma2 1 0 0 0>, <&edma2 2 0 0 0>,
-			<&edma2 3 0 1 0>, <&edma2 4 0 1 0>, <&edma2 5 0 1 0>;
+		dmas = <&edma2 0 0 0>, <&edma2 1 0 0>, <&edma2 2 0 0>,
+			<&edma2 3 0 1>, <&edma2 4 0 1>, <&edma2 5 0 1>;
 		dma-names = "rxa", "rxb", "rxc",
 				"txa", "txb", "txc";
 		fsl,asrc-rate  = <8000>;
@@ -1497,7 +1497,7 @@
 			<&clk IMX8QXP_AUD_ESAI_0_EXTAL_IPG>,
 			<&clk IMX8QXP_AUD_ESAI_0_IPG>;
 		clock-names = "core", "extal", "fsys";
-		dmas = <&edma2 6 0 1 0>, <&edma2 7 0 0 0>;
+		dmas = <&edma2 6 0 1>, <&edma2 7 0 0>;
 		dma-names = "rx", "tx";
 		power-domains = <&pd_esai0>;
 		status = "disabled";
-- 
1.7.9.5

